### PR TITLE
Update master home page with dandiset listing component

### LINKED ISSUE #244: Backport the dandiset listing component to legacy ui
Replace the current ui (girder hierarchy widget) with the newly created dandiset listing component.

This should:
- call the dandiset listing endpoints (currently will only return drafts)
- present the list of dandisets by name
- when a row is clicked, it should go to the current DLP for the dandiset
- the DLP should still go to the metadata editor
- the DLP should still allow a display of the hierarchy widget for the contents of the Dandiset folder
- the contents of the hieararchy widget should allow a link back to the DLP, using the eyeball icon as current

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Closes #244.

@mgrauer Can you take a look at the preview to check that it satisfies all of your asks?

### CODE REVIEW

**Comment by satra** [comment]
@AlmightyYakob - would it be easy to add the identifier for each dataset to the rows?

`DANDI:000004`

**Comment by AlmightyYakob** [comment]
@satra how's that look?

File: web/src/components/DandisetList.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            </v-list-item-subtitle>
+          </v-list-item-content>
+        </v-col>
+        <v-col v-if="item.meta.dandiset.identifier" cols="1">
+          <v-list-item-action>
+            <v-list-item-action-text>
+              DANDI:{{item.meta.dandiset.identifier}}
+            </v-list-item-action-text>
+          </v-list-item-action>
+        </v-col>
```
Comment (satra) [inline web/src/components/DandisetList.vue:52]:
```suggestion
            <v-list-item-subtitle>
              Contact <b>{{ getDandisetContact(item) }}</b>
              ·
              Created on <b>{{ formatDate(item.created) }}</b>
              ·
              Identifier: <b>DANDI:{{item.meta.dandiset.identifier}}</b>
            </v-list-item-subtitle>
          </v-list-item-content>
        </v-col>
```

**Comment by satra** [comment]
@AlmightyYakob - suggested a small change to where it is displayed.

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob - suggested a small change to where it is displayed.

@satra I made this change, but modified it a bit to have bolding.

**Comment by satra** [comment]
should i merge this?

**Comment by mgrauer** [comment]
I generally prefer to let the developer merge, maybe there is one other thing they want to add or they want to clean up history. For small stuff it's not as big of a deal.

**Pull request merged**